### PR TITLE
Chore: Some test type fixes

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- lots of test type fixes to remove `any`

**Why do we need this feature?**

- honestly, i'm finding there's not much value in these kinds of fixes tbh
- i think it _is_ useful to see what type things should look like (instead of just `any`), but it's a lot of boilerplate 
- anyway these are done now, but it's making me reassess whether we shouldn't just allow `any` in test files 🤷 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by itsmylife — APPROVED** [review]
Prometheus and Graphite changes look good 🚀

**Review by fabrizio-grafana — APPROVED** [review]
❤️

File: public/app/plugins/datasource/elasticsearch/QueryBuilder.test.ts [hunk 1]
```diff
@@ -19,11 +19,11 @@ describe('ElasticQueryBuilder', () => {
   it('should clean settings from null values', () => {
     const query = builder.build({
       refId: 'A',
-      // The following `missing: null as any` is because previous versions of the DS where
+      // The following `missing: null as unknown as string` is because previous versions of the DS where
       // storing null in the query model when inputting an empty string,
       // which were then removed in the query builder.
       // The new version doesn't store empty strings at all. This tests ensures backward compatibility.
-      metrics: [{ type: 'avg', id: '0', settings: { missing: null as any, script: '1' } }],
+      metrics: [{ type: 'avg', id: '0', settings: { missing: null as unknown as string, script: '1' } }],
```
Comment (matyax) [inline public/app/plugins/datasource/elasticsearch/QueryBuilder.test.ts:26]: I feel like this backwards compatibility should be reflected in the actual type of the builder and not in casting `null` as string. Unrelated with this PR but now that I see it I don't like it.

Comment (matyax) [reply public/app/plugins/datasource/elasticsearch/QueryBuilder.test.ts:26]: https://github.com/grafana/grafana/issues/81825

**Review by matyax — APPROVED** [review]
Thanks!

**Pull request merged**